### PR TITLE
feat(dropdown): remove shim and fallbacks

### PR DESCRIPTION
**Pull request by Jinnie**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: projects/angular/src/popover/dropdown/_properties.dropdown.scss [hunk 1]
```diff
@@ -5,50 +5,51 @@
...
+      --clr-dropdown-child-border-color: #{tokens.$cds-alias-object-border-color};
+      --clr-dropdown-bg-hover-color: #{tokens.$cds-alias-object-interaction-background-hover};
+      --clr-dropdown-selection-color: #{tokens.$cds-alias-object-interaction-background-selected};
       --clr-dropdown-box-shadow: var(--clr-popover-box-shadow-color);
-      --clr-dropdown-caret-icon-size: #{variables.$clr_baselineRem_0_5};
 
       // Usage:  ../data/datagrid/_datagrid.clarity.scss
       // Usage:  ../emphasis/alert/_alert.clarity.scss
-      --clr-dropdown-text-color: var(--clr-p1-color);
+      --clr-dropdown-text-color: var(--clr-global-font-color);
```
Comment (valentin-mladenov) [inline projects/angular/src/popover/dropdown/_properties.dropdown.scss:28]: For what is this used?

```suggestion
      --clr-dropdown-text-color: tokens.$cds-alias-object-interaction-color;
```

File: projects/angular/src/popover/dropdown/_properties.dropdown.scss [hunk 2]
```diff
@@ -5,50 +5,51 @@
...
 
       // Usage: ../popover/dropdown/_menu-mixins.clarity.scss
       // Dropdown Item
-      --clr-dropdown-item-color: var(--clr-color-neutral-700);
-      --clr-dropdown-item-font-weight: var(--clr-p1-font-weight);
-      --clr-dropdown-item-padding: #{variables.$clr_baselineRem_0_125 variables.$clr_baselineRem_1};
-      --clr-dropdown-font-size: inherit;
-      --clr-dropdown-line-height: inherit;
-      --clr-dropdown-letter-spacing: inherit;
+      --clr-dropdown-item-color: #{tokens.$cds-alias-typography-color-300};
```
Comment (valentin-mladenov) [inline projects/angular/src/popover/dropdown/_properties.dropdown.scss:38]:
```suggestion
      --clr-dropdown-item-color: #{tokens.$cds-alias-object-interaction-color};
```

File: projects/angular/src/popover/dropdown/_menu-mixins.clarity.scss [hunk 3]
```diff
@@ -16,110 +17,47 @@
...
-      variables.$clr-use-custom-properties
-    );
-    @include mixins.css-var(
-      color,
-      clr-dropdown-item-color,
-      dropdown-variables.$clr-dropdown-item-color,
-      variables.$clr-use-custom-properties
-    );
+    background-color: dropdown-variables.$clr-dropdown-bg-hover-color;
+    color: dropdown-variables.$clr-dropdown-item-color;
```
Comment (valentin-mladenov) [inline projects/angular/src/popover/dropdown/_menu-mixins.clarity.scss:54]: Should use different internal token based on CDS spec.

```suggestion
    color: tokens.$cds-alias-object-interaction-color-hover;
```

File: projects/angular/src/popover/dropdown/_menu-mixins.clarity.scss [hunk 4]
```diff
@@ -16,110 +17,47 @@
...
-      dropdown-variables.$clr-dropdown-selection-color,
-      variables.$clr-use-custom-properties
-    );
-    @include mixins.css-var(
-      color,
-      clr-dropdown-active-text-color,
-      dropdown-variables.$clr-dropdown-active-text-color,
-      variables.$clr-use-custom-properties
-    );
+    background-color: dropdown-variables.$clr-dropdown-selection-color;
```
Comment (valentin-mladenov) [inline projects/angular/src/popover/dropdown/_menu-mixins.clarity.scss:59]: Should it be active background color?
Or is it the other way around? 
Which color is responsible for the active (already clicked and expanded element) and selected (currently clicked element)?

It's not clear and I'm confused.
```suggestion
    background-color: dropdown-variables.$clr-dropdown-active-background-color;
```

File: projects/angular/src/utils/_mixins.scss [hunk 5]
```diff
@@ -311,7 +311,7 @@ $typography-tokens: (
     // P2, Section Header
     font-size: #{tokens.$cds-alias-typography-secondary-font-size},
     font-weight: #{tokens.$cds-alias-typography-font-weight-medium},
-    line-height: #{tokens.$cds-alias-typography-secondary-line-height},
+    line-height: #{baselinePx(24)},
```
Comment (valentin-mladenov) [inline projects/angular/src/utils/_mixins.scss:314]: Please use `#{tokens.$cds-global-space-9}` on both places. I don't know why is `baselinePx` function is used previously.

```suggestion
    line-height: #{tokens.$cds-global-space-9},
```

**Review by valentin-mladenov — COMMENTED** [review]
check my comments.

File: projects/angular/src/popover/dropdown/_dropdown.clarity.scss [hunk 6]
```diff
@@ -116,56 +109,17 @@
...
-        variables.$clr-use-custom-properties
-      );
-
-      @include mixins.css-var(
-        padding,
-        clr-dropdown-header-padding,
-        0 variables.$clr_baselineRem_0_5,
-        variables.$clr-use-custom-properties
-      );
+      @include mixins.generate-typography-token('SECONDARY-13-MD-EXP');
```
[the remark this answers is not in the document]
Comment (Jinnie) [reply projects/angular/src/popover/dropdown/_dropdown.clarity.scss:115]: Yeah, as we agreed, that will be better.

File: projects/angular/src/popover/dropdown/_menu-mixins.clarity.scss [hunk 7]
```diff
@@ -16,110 +17,47 @@
...
-  );
-  @include mixins.css-var(
-    border-width,
-    clr-global-borderwidth,
-    variables.$clr-global-borderwidth,
-    variables.$clr-use-custom-properties
-  );
+  background-color: dropdown-variables.$clr-dropdown-bg-color;
+
+  border-width: variables.$clr-global-borderwidth;
```
[the remark this answers is not in the document]
Comment (Jinnie) [reply projects/angular/src/popover/dropdown/_menu-mixins.clarity.scss:26]: Honestly, when the values are multi-line, long-string items, I prefer them separately.

File: projects/angular/src/popover/dropdown/_menu-mixins.clarity.scss (hunk 4 above)
Comment (Jinnie) [reply projects/angular/src/popover/dropdown/_menu-mixins.clarity.scss:59]: I think it's wording ambiguity. IIRC, .active here designated selected option.
But, whatever it is, I prefer to have it untangled after the mechanical task of removing the shim.
Doing the 2 together at the moment sounds a bit risky. We may file a bug in the ATM refactor epic.

File: projects/angular/src/popover/dropdown/_menu-mixins.clarity.scss [hunk 8]
```diff
@@ -16,110 +17,47 @@
...
-  @include mixins.css-var(
-    padding,
-    clr-dropdown-items-padding,
-    variables.$clr_baselineRem_0_5 0,
-    variables.$clr-use-custom-properties
-  );
+  border-color: dropdown-variables.$clr-dropdown-border-color;
+  border-radius: dropdown-variables.$clr-dropdown-border-radius;
+
+  box-shadow: 0 variables.$clr-global-borderwidth mixins.baselinePx(3) dropdown-variables.$clr-dropdown-box-shadow;
```
[the remark this answers is not in the document]
Comment (Jinnie) [reply projects/angular/src/popover/dropdown/_menu-mixins.clarity.scss:31]: Great!

File: projects/angular/src/popover/dropdown/_dropdown.clarity.scss [hunk 9]
```diff
@@ -242,34 +180,18 @@
...
 
     .dropdown-divider {
-      font-size: dropdown-variables.$clr-dropdown-header-font-size;
-      @include mixins.css-var(
-        border-bottom-width,
-        clr-dropdown-divider-border-width,
-        variables.$clr-global-borderwidth,
-        variables.$clr-use-custom-properties
-      );
+      border-bottom-width: dropdown-variables.$clr-dropdown-divider-border-width;
```
[the remark this answers is not in the document]
Comment (Jinnie) [reply projects/angular/src/popover/dropdown/_dropdown.clarity.scss:183]: Same as below - when the values are multi-line, long-string items, I prefer them separately for better readability.
`1px solid pink` is quite different from `dropdown-variables.$clr-dropdown-divider-border-width solid dropdown-variables.$clr-dropdown-divider-color`

File: projects/angular/src/popover/dropdown/_dropdown.clarity.scss [hunk 10]
```diff
@@ -54,9 +55,11 @@
         --cds-global-space-7: #{dropdown-variables.$clr-dropdown-caret-icon-dimension};
       }
 
+      $clr-dropdown-caret-left-margin: tokens.$cds-global-space-2;
       &.btn {
         padding-right: calc(
-          #{button-variables.$clr-btn-horizontal-padding} + #{dropdown-variables.$clr-dropdown-caret-icon-dimension} + #{dropdown-variables.$clr-dropdown-caret-left-margin}
+          button-variables.$clr-btn-horizontal-padding + dropdown-variables.$clr-dropdown-caret-icon-dimension +
+            $clr-dropdown-caret-left-margin
```
[the remark this answers is not in the document]
Comment (Jinnie) [reply projects/angular/src/popover/dropdown/_dropdown.clarity.scss]: Added it to all calc occurrences.

File: projects/angular/src/popover/dropdown/_menu-mixins.clarity.scss [hunk 11]
```diff
@@ -16,110 +17,47 @@
...
-  );
-  font-size: dropdown-variables.$clr-dropdown-item-font-size;
-  @include mixins.css-var(
-    font-weight,
-    clr-dropdown-item-font-weight,
-    dropdown-variables.$clr-dropdown-item-font-weight,
-    variables.$clr-use-custom-properties
-  );
-  letter-spacing: dropdown-variables.$clr-dropdown-item-letter-spacing;
+  @include mixins.generate-typography-token('SECONDARY-13-RG-EXP');
```
[the remark this answers is not in the document]
Comment (Jinnie) [reply projects/angular/src/popover/dropdown/_menu-mixins.clarity.scss:40]: added all the typography vars and props back

File: projects/angular/src/popover/dropdown/_menu-mixins.clarity.scss (hunk 3 above)
Comment (Jinnie) [reply projects/angular/src/popover/dropdown/_menu-mixins.clarity.scss:54]: done

File: projects/angular/src/popover/dropdown/_properties.dropdown.scss [hunk 12]
```diff
@@ -5,50 +5,61 @@
...
 
       // Usage: ../popover/dropdown/_menu-mixins.clarity.scss
       // Dropdown Item
-      --clr-dropdown-item-color: var(--clr-color-neutral-700);
-      --clr-dropdown-item-font-weight: var(--clr-p1-font-weight);
-      --clr-dropdown-item-padding: #{variables.$clr_baselineRem_0_125 variables.$clr_baselineRem_1};
-      --clr-dropdown-font-size: inherit;
-      --clr-dropdown-line-height: inherit;
-      --clr-dropdown-letter-spacing: inherit;
+      --clr-dropdown-item-color: #{tokens.$cds-alias-object-interaction-color};
```
Comment (valentin-mladenov) [inline projects/angular/src/popover/dropdown/_properties.dropdown.scss:42]:
```suggestion
      --clr-dropdown-item-color: var(-clr-dropdown-text-color);
```

File: projects/angular/src/popover/dropdown/_properties.dropdown.scss [hunk 13]
```diff
@@ -5,50 +5,61 @@
...
+      --clr-dropdown-child-border-color: #{tokens.$cds-alias-object-border-color};
+      --clr-dropdown-bg-hover-color: #{tokens.$cds-alias-object-interaction-background-hover};
+      --clr-dropdown-selection-color: #{tokens.$cds-alias-object-interaction-background-selected};
       --clr-dropdown-box-shadow: var(--clr-popover-box-shadow-color);
-      --clr-dropdown-caret-icon-size: #{variables.$clr_baselineRem_0_5};
 
       // Usage:  ../data/datagrid/_datagrid.clarity.scss
       // Usage:  ../emphasis/alert/_alert.clarity.scss
-      --clr-dropdown-text-color: var(--clr-p1-color);
+      --clr-dropdown-text-color: tokens.$cds-alias-object-interaction-color;
```
Comment (valentin-mladenov) [inline projects/angular/src/popover/dropdown/_properties.dropdown.scss:28]:
```suggestion
      --clr-dropdown-text-color: #{tokens.$cds-alias-object-interaction-color};
```

File: projects/angular/src/popover/dropdown/_menu-mixins.clarity.scss (hunk 4 above)
Comment (Jinnie) [reply projects/angular/src/popover/dropdown/_menu-mixins.clarity.scss:67]: Refactoring task created https://jira.eng.vmware.com/browse/CDE-1422

**Review by valentin-mladenov — APPROVED** [review]
LGTM

**Pull request merged**